### PR TITLE
Grammar notes update for Swift implementation

### PR DESCRIPTION
**Pull request by yuryleb**
This PR updates Grammar feature description with notes about [OSRM Text Instructions for Swift](https://github.com/Project-OSRM/osrm-text-instructions.swift/pull/49) implementation specifics - always global matching regardless of specified regular expressions flags.

Some Russian regular expressions are also joint and optimized to work better in Swift too.

### CODE REVIEW

**Comment by mcwhittemore** [comment]
@1ec5 can you take a look at this?

File: Grammar.md [hunk 1]
```diff
@@ -28,9 +28,21 @@ The required grammatical case should be specified right in instruction's substit
 - Instruction text formatter ([index.js](index.js) in this module) should:
   - check `{way_name}` and `{rotary_name}` variables for optional grammar case after colon: `{way_name:accusative}`
   - find appropriate regular expressions block for target language and specified grammar case
-  - call standard [string replace with regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) for each expression in block passing result from previous call to the next; the first call should enclose original street name with whitespaces to make parsing words in names a bit simplier.
+  - call standard [string replace with regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) for each expression in block passing result from previous call to the next; the first call should enclose original street name with whitespaces to make parsing several words inside name a bit simplier.
```
Comment (1ec5) [inline Grammar.md]: Is this still a requirement, now that `grammarize` is called before capitalization and insertion (Project-OSRM/osrm-text-instructions#170)?

**Comment by jcsg** [comment]
@1ec5 Yes, the changes to the grammar rules look good to me.

**Comment by 1ec5** [comment]
Thanks @yuryleb!

**Pull request merged**